sources/web/datalab/polymer/components/datalab-app/datalab-app.ts

```diff
@@ -55,6 +55,14 @@ class DatalabAppElement extends Polymer.Element {
 
     // Will be called after the custom element is done rendering.
     window.addEventListener('WebComponentsReady', this._boundResizeHandler, true);
+
+    ApiManager.disconnectedHandler = () => {
```

Comment:
I do not have a specific scenario in mind. Your addition of the isConnected flag helps to reduce the possibility of race conditions in clearing the toast. I am OK with moving ahead with it as is.